ddtrace/tracer/option.go

```diff
@@ -154,6 +154,12 @@ type config struct {
 
 	// spanAttributeSchemaVersion holds the selected DD_TRACE_SPAN_ATTRIBUTE_SCHEMA version.
 	spanAttributeSchemaVersion int
+
+	// peerServiceDefaultsEnabled indicates whether the peer.service tag calculation is enabled or not.
+	peerServiceDefaultsEnabled bool
+
+	// peerServiceMappings holds a set of service mappings to dynamically rename peer.service values.
+	peerServiceMappings map[string]string
```

Comment:
Would it be possible to add `tracer.StartOption` for these new configurations as well as the schema version? Internally, we have a pattern of leveraging consulconfig to enable new features application-by-application using common bootrapping code for the tracer. This requires us having a programmatic way of enabling these features.

Reply:
Definitely! let's discuss with @katiehockman and/or @ajgajg1134 what they think about this and if it's ok I can open a follow-up PR adding options for all the recently added configurations, including the naming schema version.

Reply:
Sure I see no reason we shouldn't do this. A follow-up PR would be good since this was is already pretty large.

Reply:
It looks like the options are already present in this PR. It would be great if we could keep them in given the urgency around getting this merged and deployed to our internal services.